src/Symfony/Component/Serializer/Tests/SerializerTest.php

```diff
@@ -1247,6 +1248,29 @@ public function provideCollectDenormalizationErrors()
             [new ClassMetadataFactory(new AnnotationLoader(new AnnotationReader()))],
         ];
     }
+
+    public function testCacheableNormalizer()
+    {
+        $normalizer = $this->createMock(NormalizerInterface::class);
+        $serializer = new Serializer([$normalizer], []);
+
+        $normalizer
+            ->expects($this->exactly(3))
+            ->method('supportsNormalization')
+            ->willReturnCallback(function ($data, $format, array $context = []): CacheableSupport {
+                if (!$data instanceof Bar) {
+                    return CacheableSupport::SupportNever;
+                }
+
+                return ($context['TEST_CONTEXT'] ?? false) ? CacheableSupport::Support : CacheableSupport::SupportNot;
+            });
+
+        $this->assertTrue($serializer->supportsNormalization(new Bar(''), 'json', ['TEST_CONTEXT' => true]));
+        $this->assertFalse($serializer->supportsNormalization(new Bar(''), 'json'));
+        $this->assertFalse($serializer->supportsNormalization(new \StdClass(), 'json', ['TEST_CONTEXT' => true]));
+        $this->assertFalse($serializer->supportsNormalization(new \StdClass(), 'json', ['TEST_CONTEXT' => true]));
+        $this->assertFalse($serializer->supportsNormalization(new \StdClass(), 'json', ['TEST_CONTEXT' => true]));
+    }
```

Comment:
This is testing that, by calling `supports..`  multiple times and ensuring `supportsNormalization` is only called 3 times instead of 5. 

```
$normalizer
            ->expects($this->exactly(3))
            ->method('supportsNormalization')
```

Unless you'd say something more explicit could be added?